bots/make-checkout

```diff
@@ -52,24 +53,31 @@ def main():
             sys.stderr.write("> " + output + "\n")
         return output
 
-    execute("git", "fetch", "origin", opts.ref)
+    if opts.repo:
+        if "test\n" in subprocess.check_output([ "git", "remote" ], universal_newlines=True):
+            execute("git", "remote", "remove", "test")
+        execute("git", "remote", "add", "test", "https://github.com/" + opts.repo)
+
+    execute("git", "fetch", "test" if opts.repo else "origin", opts.ref)
     execute("git", "checkout", "--detach", opts.revision)
 
-    # COMPAT: If the bots directory doesn't exist in this branch, check it out from master
-    if opts.base and opts.base != "master":
-        sys.stderr.write("Checking out bots directory from master ...\n")
+    # If the bots directory doesn't exist in this branch or repo, check it out from master
+    if opts.repo or (opts.base and opts.base != "master"):
+        sys.stderr.write("Checking out bots directory from Cockpit master ...\n")
         execute("git", "checkout", "--force", "origin/master", "--", "bots/")
 
-        # The machine code is special, copy it from master too
+        # The machine code has symlinks, replace them with regular files
         machine = os.path.join(BOTS, "machine")
         for name in os.listdir(machine):
             path = os.path.join(machine, name)
             if os.path.islink(path):
                 os.unlink(path)
-                code = subprocess.check_output([ "git", "show", "cockpit/master:test/common/{0}".format(name) ],
+                code = subprocess.check_output([ "git", "show", "origin/master:test/common/{0}".format(name) ],
                                                cwd=BASE)
                 with open(path, "wb") as f:
                     f.write(code)
 
+        execute("git", "reset", "HEAD")
```

Comment:
What does this do? My gut feeling is that a `reset --hard` and a `clean -fdx` should be done after the `git checkout` of opts.revision, before checking out bots/, to ensure that there are no leftovers in the tree.

Reply:
This resets the index so that the subsequent rebase in test-invoke can start.  But I don't know why we didn't need this before, so I should definitely investigate this more.

Reply:
I guess we just didn't do any rebases on PRs for the non-master branches, which sounds reasonable since there are so few of them.

I think it is proper for `make-checkout` to exit with a clean index, but we can also move the "git reset HEAD" to tests-invoke. Wdyt?